src/events/mysql.js

```diff
@@ -15,14 +15,17 @@ function mysqlQueryWrapper(wrappedFunction) {
         let queryString;
         let callback;
         let params;
-        // This function can be called with either a string or Query object
-        if (typeof sql === 'string') {
-            queryString = sql;
-            ({ params, callback } = sqlWrapper.parseQueryArgs(arg1, arg2));
-        } else {
+        if (typeof sql !== 'string') {
```

Comment:
I think there is a little issue with that:
We have seen that in 'sequalize', the object is not `Commands.Query` but still has an `sql` field containing the query.
Although, it may not have the `values` field set, and both `values` and `callback` can be given as arguments (see `mysql2.connection.createQuery`). But you only take `callback` argument in this case.
Seems to me that the correct thing to do is to check if `onResult` is set, and decide by it wether to take both arguments from the `sql` object or from this function's arguments/

Reply:
But sometimes `sql.onResult` is undefined, and `sql.sql` is defined.